libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java

```diff
@@ -486,35 +479,6 @@ public void onSessionStateChanged(@NotNull TripSessionState tripSessionState) {
     }
   };
 
-  private void updateRoute(DirectionsRoute route) {
```

Comment:
looks like this is how legacy was working, but as a public function

https://github.com/mapbox/mapbox-navigation-android/blob/be62ff91c2bf2a0d09bb0fe1f1ffc71f710da33b/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java#L259-L268

Reply:
Legacy was different, faster route was disabled by default and if enabled we always updated the route on behalf the client, they didn't have the option to update the route externally it was an all or nothing, we handle that internally (`updateRoute` was package-private 😅) with that flag (`enableFasterRouteDetection`).